test/verify/quant_conv_default_mode.cpp

```diff
@@ -10,9 +10,9 @@ struct quant_conv_default_mode : verify_program<quant_conv_default_mode>
     {
         migraphx::program p;
         auto* mm = p.get_main_module();
-        migraphx::shape a_shape{migraphx::shape::int8_type, {2, 3, 4, 4}};
+        migraphx::shape a_shape{migraphx::shape::int8_type, {16, 16, 4, 4}};
```

Comment:
Why are these tests changed?

Reply:
@pfultz2 `[2, 3, 4, 4]` dimensions will be converted to `[2, 4, 4, 4]` from `miopen_quant_convolution::pack_int8_shape()`, which happens to be supported by `int8x4`. 

On the other hand, neither `[2, 3, 4, 4]` nor `[2, 4, 4, 4]` are supported dimensions with `int8`, so I manually picked a dimension small enough and still supported.

Reply:
We shouldn't change the original test as we need to ensure such sizes will work(or do you mean these tests will break?).

We can add a test with the `16, 16, 4, 4` input size to get test coverage.

Reply:
Correct, those tests will break with previous dimensions: they will end with "MIOpen Error: /root/MIOpen/src/ocl/convolutionocl.cpp:518: Forward Convolution cannot be executed due to incorrect params".

Also I'm curious on why does MIGraphX care on a specific set of dimension? I'm assuming as long as it is supported by the library under the hood, it proves the MIGraphX correctness. Whatever that doesn't get supported is a MIOpen problem, not MIGraphX problem.

Reply:
> Correct, those tests will break with previous dimensions 

Then we need to add more logic in the pack_int8_args to check for the dimensions supported by miopen. 

> Also I'm curious on why does MIGraphX care on a specific set of dimension?

We can have a model fail to run just because it didn't use dimensions that was nice for MIOpen.

Reply:
> We cannot have a model fail to run just because it didn't use dimensions that was nice for MIOpen.

Fair enough.

> Then we need to add more logic in the pack_int8_args to check for the dimensions supported by miopen.

Are you talking about checking whether MIOpen supports int8 with current config, and use int8 only when it is supported? (otherwise fallback to int8x4). From what I can see that's the job of the `quant_convolution.cpp`, that it only knows whether the config can be supported after the `miopenFindConvolutionForwardAlgorithm()` call. But then `pack_int_args` seems to be just a lowering pass and making it to invoke the miopen API seems to be beyond its responsibility (or capability, because it mostly have just the shape information, if I understand it correctly.

Reply:
> Are you talking about checking whether MIOpen supports int8 with current config, and use int8 only when it is supported? (otherwise fallback to int8x4).

Yes.

> From what I can see that's the job of the quant_convolution.cpp, that it only knows whether the config can be supported after the miopenFindConvolutionForwardAlgorithm() call.

Do you mean when we call `compute` method for quant_convolution? If find fails in compute(or finalize), then we cannot fallback on int8x4 since the input is a different layout.

> But then pack_int_args seems to be just a lowering pass and making it to invoke the miopen API seems to be beyond its responsibility (or capability, because it mostly have just the shape information, if I understand it correctly.

Ah thats right, the logic needs to be in lowering not in pack_int_args(which just checks the `int8_x4_format` flag). In lowering we already do call find so we can get the workspace:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/targets/gpu/lowering.cpp#L369

We just need to put a try/catch around this and fallback on int8x4 if it throws an error.

Reply:
@pfultz2 For sure. Test case wise, which of the following do you prefer:
 - Add an additional set of test with {16,4,4,4} to explicitly regress int8 shape; preserve the {2,3,4,4} to test the fallback
 - Mix both {16,4,4,4} and {2,3,4,4} between different test cases such that we have coverage for both (and it reduced duplication)

Reply:
Done in 3d972e24d5f40201252765b5c8c6c611beda8349